### PR TITLE
add default backend configurable flag support

### PR DESCRIPTION
**Pull request by pgvishnuram**
## Description

This PR introduces a new flag to control nginx default backend to be controlled by end user whether or not this can be deployed.

Note: this is not a breaking change - by disabling this default backend we will lose the default error pages 

## Related Issues

- https://github.com/astronomer/issues/issues/6748

## Testing

Need to update testing info

## Merging
cherry-pick to release-0.36

### CODE REVIEW

File: charts/nginx/templates/nginx-default-backend-networkpolicy.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ######################################
 ## Nginx controller pods NetworkPolicy
 ######################################
-{{- if .Values.global.networkPolicy.enabled }}
+{{- if and .Values.defaultBackend.enabled  .Values.global.networkPolicy.enabled }}
```
Comment (danielhoherd) [inline charts/nginx/templates/nginx-default-backend-networkpolicy.yaml:4]:
```suggestion
{{- if and .Values.defaultBackend.enabled .Values.global.networkPolicy.enabled }}
```

File: charts/nginx/templates/nginx-default-backend-pod-disruption-budget.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 #########################################################
 ## Astronomer Nginx Default Backend Pod Disruption Budget
 #########################################################
-{{- if .Values.global.podDisruptionBudgetsEnabled }}
+{{- if and .Values.defaultBackend.enabled  .Values.global.podDisruptionBudgetsEnabled }}
```
Comment (danielhoherd) [inline charts/nginx/templates/nginx-default-backend-pod-disruption-budget.yaml:4]:
```suggestion
{{- if and .Values.defaultBackend.enabled .Values.global.podDisruptionBudgetsEnabled }}
```

File: charts/nginx/templates/nginx-default-backend-serviceaccount.yaml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 ##########################
 ## NGINX ServiceAccount ##
 ##########################
-{{- if .Values.global.rbacEnabled }}
+{{- if and .Values.defaultBackend.enabled  .Values.global.rbacEnabled }}
```
Comment (danielhoherd) [inline charts/nginx/templates/nginx-default-backend-serviceaccount.yaml:4]:
```suggestion
{{- if and .Values.defaultBackend.enabled .Values.global.rbacEnabled }}
```

**Review by rishkarajgi — CHANGES_REQUESTED** [review]
needs tests

**Review by rishkarajgi — APPROVED** [review]

**Pull request merged**